packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -182,6 +183,10 @@ type InternalState<M extends BaseMetadata> = Readonly<{
   queries3: Record<string, QueryAsyncResult>; // Notification settings
   queries4: Record<string, QueryAsyncResult>; // Versions
 
+  threads: {
+    loadThreadsPromises: Record<string, Promise<ThreadData<M>[]>>;
```

Comment:
We may also need to create a new separate field to track the promise status because we for the suspense `useThreads` hook, we can expect the `use` polyfill to add the `status` property on the promise but the non-suspense hook doesn't use the `use` API.

Reply:
Actually I had a thought about this this morning! Could we create our own promise wrapper, that would add the `use()`-compatible `status` fields?

```tsx
const coolerPromise = pinkyPromise(originalPromise);
coolerPromise.status // 'pending'
await coolerPromise;
coolerPromise.status // 'resolved'
```

This way we can synchronously read the promise's status (which is useful for us anyway), and it will additionally be compatible with `use()` already, so `use()` will never have to patch anything.

Reply:
In places like this record, we could simply type the values part as a `PinkyPromise`:

```tsx
    loadThreadsPromises: Record<string, PinkyPromise<ThreadData<M>[]>>;
```

Reply:
The implementation would be almost the same as `use()`, except it would not _throw_.